examples/2d/tilemap_chunk.rs

```diff
@@ -0,0 +1,88 @@
+//! Shows a tilemap chunk rendered with a single draw call.
+
+use bevy::{
+    dev_tools::fps_overlay::FpsOverlayPlugin,
+    prelude::*,
+    sprite::{TilemapChunk, TilemapChunkIndices},
+};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins.set(ImagePlugin::default_nearest()),
+            FpsOverlayPlugin::default(),
+        ))
+        .add_systems(Startup, setup)
+        .add_systems(Update, (spawn_tilemap, update_tilemap))
+        .run();
+}
+
+#[derive(Resource)]
+struct LoadingTileset {
+    is_loaded: bool,
+    handle: Handle<Image>,
+}
+
+#[derive(Component, Deref, DerefMut)]
+struct UpdateTimer(Timer);
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.insert_resource(LoadingTileset {
+        is_loaded: false,
+        handle: asset_server.load("textures/array_texture.png"),
+    });
+
+    commands.spawn(Camera2d);
+}
+
+fn spawn_tilemap(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut loading_tileset: ResMut<LoadingTileset>,
+    mut images: ResMut<Assets<Image>>,
+) {
+    if loading_tileset.is_loaded
```

Comment:
Instead of doing this is_loaded check, this system should be in Startup so it runs only once.

Reply:
For this one I'm mirroring the [array_texture example](https://github.com/bevyengine/bevy/blob/main/examples/shader/array_texture.rs). I figured this was required because we need to create a new Image with `reinterpret_stacked_2d_as_array`.

Reply:
Oooh, yeah, I misunderstood what was going on there.

In that case, I'd suggest rewriting this to use events. There's gAssetEvent::LoadedWithDependencies` that should work here. 

I'd also suggest everything that doesn't need the image data to startup. So the spawning of the tilechunk could be in startup. It shouldn't matter if the asset isn't loaded when it spawns, it should just wait until the asset is loaded before rendering.

Reply:
Would that look like making `tileset` Optional, or maybe separating Tileset out into it's own component?

Reply:
No, you can just use the handle you get from the asset server from the startup. That material system will simply not process the material using the handle until it is loaded.

Reply:
But the handle we get in startup is for the atlas image, not the tileset layer image. I think this will be nicer once we have asset processing because you can just load something like a tileset ron file. That's what I have prototyped here: https://github.com/ConnerPetzold/bevy_tilemap/blob/main/examples/basic.rs

Reply:
Sorry yeah I see what you're saying now, I forgot that it's using the same handle. Let me give it a try. I'm not sure what will happen if the material tries to use the asset image before it's reinterpreted as an array.